cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -410,82 +410,91 @@ public StreamConfig create(final StreamId streamId, @Nullable final Properties p
     // User should have write access to the namespace
     NamespaceId streamNamespace = streamId.getParent();
     ensureAccess(streamNamespace, Action.WRITE);
-    // revoke privileges to make sure there is no orphaned privileges
-    privilegesManager.revoke(streamId);
+
     final Properties properties = (props == null) ? new Properties() : props;
+    String ownerPrincipal = properties.containsKey(Constants.Security.PRINCIPAL) ?
+      properties.getProperty(Constants.Security.PRINCIPAL) : null;
+
+    if (exists(streamId)) {
+      // if stream exists then make sure owner for this create request is same
+      verifyOwner(streamId, ownerPrincipal);
+      // stream create is an idempotent operation as of now so just return null and don't do anything
+      return null;
```

Comment:
OK. Seems like the old code is returning null as well. In that case, can you mark the method a `@Nullable` in the `StreamAdmin` interface (for both create methods) and document when it'll return null?